src/vs/workbench/api/common/extHostTelemetry.ts

```diff
@@ -275,7 +278,9 @@ export class ExtHostTelemetryLogger {
 
 	dispose(): void {
 		if (this._sender?.flush) {
-			this._sender.flush();
+			// Disposed of so now we set it to undefined
+			Promise.resolve(this._sender.flush()).then(this._sender = undefined);
```

Comment:
Should `this._sender` be unset synchronously so any callers immediately see the "disposed" behavior?

Reply:
👍 

also `this._sender` will stick around when flushing fails (for whatever reason)